### PR TITLE
[MODINVSTOR-310] Add date and source fields to the circulation notes

### PR DESCRIPTION
**Pull request by opochepam**
## Purpose
Resolves: MODINVSTOR-310
Add date and source fields to the circulation notes
## Links
https://issues.folio.org/browse/MODINVSTOR-310
https://github.com/folio-org/mod-inventory/pull/125

### CODE REVIEW

File: ramls/item.json [hunk 1]
```diff
@@ -128,6 +132,34 @@
             "type": "string",
             "description": "Text to display"
           },
+          "source": {
+            "type": "object",
+            "description": "The user who added the note",
```
Comment (marcjohnson-kint) [inline ramls/item.json]: Is this always the user who added the note, or does it change when the note is updated (to the user who last updated the note)?

File: ramls/item.json [hunk 2]
```diff
@@ -128,6 +132,34 @@
...
+            "properties": {
```
Comment (marcjohnson-kint) [inline ramls/item.json]: As these properties are stored copies of information from the user (presumably to support sorting), they should be marked read only, as this behaviour is controlled by the server.

Comment (marcjohnson-kint) [inline ramls/item.json]: As these properties are intended to be owned by the server, I think it could be useful to reflect that in the description

Comment (opochepam) [reply ramls/item.json]: Updated descriptions for the user and the date.

File: ramls/item.json [hunk 3]
```diff
@@ -128,6 +132,34 @@
             "type": "string",
             "description": "Text to display"
           },
+          "source": {
+            "type": "object",
+            "description": "The user who added/updated the note. The property is generated by the server and needed to support sorting.",
+            "properties": {
+              "id": {
+                "type": "string",
+                "description": "The id of the user who added/updated the note"
```
Comment (julianladisch) [inline ramls/item.json]: Does this point to /users/{id}, if yes, then it should be mentioned in the description.

File: ramls/item.json [hunk 4]
```diff
@@ -128,6 +132,34 @@
             "type": "string",
             "description": "Text to display"
           },
+          "source": {
+            "type": "object",
+            "description": "The user who added/updated the note. The property is generated by the server and needed to support sorting.",
+            "properties": {
+              "id": {
```
Comment (julianladisch) [inline ramls/item.json]: This is the 3rd "id" field in this file. Can you add a description to the "id" field in line 6 to explain the difference?

File: ramls/item.json (hunk 3 above)
Comment (opochepam) [reply ramls/item.json]: Yes it does. Updated the description.

File: ramls/item.json (hunk 4 above)
Comment (opochepam) [reply ramls/item.json]: Added the description.

File: ramls/item.json (hunk 3 above)
Comment (julianladisch) [reply ramls/item.json]: Thanks for adding it to the description of "source" in line 138, can you also add it to the "id" description in line 142?

Comment (opochepam) [reply ramls/item.json]: done

**Pull request merged**